jobs/shield-daemon/templates/config/shieldd.conf

```diff
@@ -1,40 +1,75 @@
+  else
+    l = link('shield-db')
+    scheme = l.p('db_scheme')
+    port =   l.p('port')
+    host =   l.instances.first.address
+    dbname = l.p('databases').select {|database| database['tag'] == 'shield'}.first['name']
+    role =   l.p('roles').first
+    username, password = role['name'], role['password']
   end
-%>
+  if scheme == "mysql"
+    shielddb_uri = "#{username}:#{password}@tcp(#{host}:#{port})/#{dbname}?sql_mode=POSTGRESQL&default_storage_engine=INNODB"
+  else
+    shielddb_uri = "#{scheme}://#{username}:#{password}@#{host}:#{port}/#{dbname}?sslmode=disable"
+  end
+  shielddb_scheme = scheme
+
+  if_p("auth.oauth.provider") do
+    if p('database.host', nil)
+      username, password, host, port, db = p('auth.oauth.sessions.db.username'), p('auth.oauth.sessions.db.password'), p('auth.oauth.sessions.db.host'), p('auth.oauth.sessions.db.port'), p('auth.oauth.sessions.db.name')
+      scheme = p('auth.oauth.sessions.db.type', 'postgres')
+    else
+      l = link('sessions-db')
+      scheme = l.p('db_scheme')
+      port =   l.p('port')
+      host =   l.instances.first.address
+      dbname = l.p('databases').select {|database| database['tag'] == 'sessions'}.first['name']
+      role =   l.p('roles').first
+      username, password = role['name'], role['password']
+    end
+    if scheme == "mysql"
+      sessionsdb_uri = "#{username}:#{password}@tcp(#{host}:#{port})/#{db}?sql_mode=POSTGRESQL&default_storage_engine=INNODB"
+    else
+      sessionsdb_uri = "#{scheme}://#{username}:#{password}@#{host}:#{port}/#{db}?sslmode=disable"
+    end
+    sessionsdb_scheme = scheme
+  end
+-%>
 ---
-database_type: <%= p("shield.daemon.database.type") %>
-database_dsn: "<%= database_dsn %>"
-listen_addr: 127.0.0.1:8080
+database_type: "<%= shielddb_scheme %>"
+database_dsn: "<%= shielddb_uri %>"
+listen_addr: 0.0.0.0:8080
 private_key: /var/vcap/jobs/shield-daemon/shared/id_rsa
 web_root: /var/vcap/packages/shield/webui
 
 
 auth:
-<% if_p("shield.daemon.auth.api_keys") do |tokens| %>
+  basic:
+    user: <%= p("auth.username") %>
+    password: <%= p("auth.password") %>
+<% if_p("auth.api_keys") do |tokens| -%>
   api_tokens:
-<% tokens.each do |k, v| %>
+<% tokens.each do |k, v| -%>
     <%= k %>: <%= v %>
-<% end %>
-<% end %>
-  basic:
-    user: <%= p("shield.daemon.auth.basic_user") %>
-    password: <%= p("shield.daemon.auth.basic_password") %>
-<% if_p("shield.daemon.auth.oauth.provider") do |provider| %>
+<% end -%>
+<% end -%>
+<% if_p("auth.oauth.provider") do |provider| -%>
   oauth:
     provider: <%= provider %>
-    key: <%= p("shield.daemon.auth.oauth.key") %>
-    secret: <%= p("shield.daemon.auth.oauth.secret") %>
-    base_url: https://<%= p("shield.daemon.domain") %>:<%= p("shield.daemon.port") %>
+    key: <%= p("auth.oauth.key") %>
+    secret: <%= p("auth.oauth.secret") %>
+    base_url: https://<%= p("domain", link("shield-daemon").instances.first.address) %>:<%= p("port") %>
```

Comment:
Yeah, hopefully it works like that. I've only tested the link usage. I kept `domain` as I assumed ppl were using it for external routes/https. I will add an operations file to do route registrar and set `domain`